docs/ref/models/fields.txt

```diff
@@ -820,6 +820,13 @@ The default form widget for this field is a single
 :class:`~django.forms.DateTimeInput`. The admin uses two separate
 :class:`~django.forms.TextInput` widgets with JavaScript shortcuts.
 
+.. warning::
+
+    ``datetime.date`` instances used with :class:`DateTimeField` will use
+    midnight as their time component. This is true for both storage and
+    comparison. To compare the date portion of a :class:`DateTimeField` with a
+    ``datetime.date`` instance, use the :lookup:`date` lookup.
+
```

Comment:
My 2¢: this warning is only necessary for `USE_TZ = False` so not sure it's worth it 🤔

If `USE_TZ = True`, Django will emit a Python warning due to the naive timestamp (and fill in the missing timezone with the default timezone).

Reply:
@shangxiao A `datetime.date` will be treated as a `datetime.datetime` with zeroed-out time regardless of the `USE_TZ` settings's value, correct? I may be misunderstanding how `USE_TZ` is related here.

I'm understanding the ticket to be asking for a warning for users that may not be aware of the "precision loss" caused by mixing these types in the ORM. Is that how you're reading it?

For what it's worth, I personally agree that the warning is not worth it.